plugins/pi4j-plugin-linuxfs/src/main/java/com/pi4j/plugin/linuxfs/provider/spi/LinuxFsSpi.java

```diff
@@ -393,6 +370,75 @@ public int write(byte[] data, int offset, int length) {
         return length;
     }
 
+    @Override
+    public int writeRead(byte[] write, int writeOffset, int writeNumberOfBytes, short writeDelayUsec, byte[] read, int readOffset, int readNumberOfBytes, short readDelayUsec) {
+        final int firstRecord = 0;
+        final int secondRecord = 1;
+        final int totalRecords = 2;
+
+        Objects.checkFromIndexSize(writeOffset, writeNumberOfBytes, write.length);
+        Objects.checkFromIndexSize(readOffset, readNumberOfBytes, read.length);
+        spi_ioc_transfer[] transferArray = getTransferArray(totalRecords);
+        PeerAccessibleMemory txBuf = new PeerAccessibleMemory(readNumberOfBytes);
+        PeerAccessibleMemory rxBuf = new PeerAccessibleMemory(readNumberOfBytes);
+
+           //    PeerAccessible
+      //  Memory writeBuf = peerArray[firstRecord];//new PeerAccessibleMemory(SPI_BUFFSIZ);
+        txBuf.write(0, write, 0, writeNumberOfBytes);
+        spi_ioc_transfer txEntry = transferArray[firstRecord]; //new spi_ioc_transfer();
+
+        // set fields in the tx transfer msg
+        txEntry.tx_buf = txBuf.getPeer();
+        txEntry.rx_buf = 0;
+        txEntry.bits_per_word = BITS8;
+        txEntry.speed_hz = config.baud();
+        txEntry.delay_usecs = writeDelayUsec;
+        txEntry.cs_change = 0;
+        txEntry.len = writeNumberOfBytes;
+
+          // RX entry
+        spi_ioc_transfer rxEntry = transferArray[1];
+        rxEntry.rx_buf = rxBuf.getPeer();
+        rxEntry.tx_buf = 0;
+        rxEntry.bits_per_word = BITS8;
+        rxEntry.speed_hz = config.baud();
+        rxEntry.delay_usecs = readDelayUsec;
+        rxEntry.cs_change = 0;
+        rxEntry.len = readNumberOfBytes;
+
+        int ret = libc.ioctl(fd, SPI_IOC_MESSAGE(totalRecords), (Object) transferArray);
+
+        logger.trace("[SPI::RegRead] <- Number bytes read {} ", ret);
+        if (ret < 0) {
+            logger.error("Failed ioctl SPI message. ret {}, error: {}", ret, Native.getLastError());
+        }
+        rxBuf.read(0, read, readOffset, readNumberOfBytes);
+
+        return readNumberOfBytes;
```

Comment:
As this is just the input parameter, can we make the method void?

Returning a value here might mislead users to try to put this into a loop, as one would do with stream read operations....